src/users_db.rs

```diff
@@ -54,30 +54,30 @@ impl UserBuilder {
     pub fn new() -> UserBuilder {
         UserBuilder {
             id: None,
-            name: "".to_owned(),
-            email: "".to_owned(),
-            password: "".to_owned(),
-            secret: "".to_owned(),
+            name: String::new(),
+            email: String::new(),
+            password: String::new(),
+            secret: String::new(),
             error: None,
             is_admin: Some(false)
         }
     }
 
-    pub fn id(&mut self, id: i32) -> &mut UserBuilder {
+    pub fn id(mut self, id: i32) -> Self {
```

Comment:
First, if `self` is the shortcut for the receiver, then `Self` is the shortcut for _the type of `self`_ which is `UserBuilder`.

Now, this is a big change and not trivial. What happens is as follows: for the sake of chaining, if we accept a mutable reference then we can not return the value (we were moving the value out of borrowed content, i.e. we were trying to create two modification points which is forbidden in Rust) so we are forced to return a mutable reference. If so, in `finalize()` we are **forced** to perform a clone to give the new User value the data it must own. This makes the API to perform worse than the current solution.

In the current solution, we chain by owning and returning the ownership. So, the access point is moved along the chain. This works fine while we don't try to access other _chainable_ methods inside any of them (more about this later). This give us an important advantage over the previous strategy. Now, in `finalize()` we simply move the proper data into their final destination inside the User structure.

This makes sense as a Builder acts somehow like a proxy for the structure being built. The Builder owns the same resources that will become the User resources. That's the idea. Avoiding unnecessary conversions in the middle. :)

But what happen in the side of the consumer, because with the former approach I could do:
```rust
let me = Builder::new().name("Salva").finalize();
```
And now I'm forced to do:
```rust
let me = Builder::new().name(String::from("Salva")).finalize();
```
Well, that's what happen in tests or in examples where your strings are not provided by outer sources so you'll end using stack or static allocated strings (i.e. &str) where in the wild, you would be dealing with heap allocated strings (i.e. Strings). So I prefer for my API to be as efficient as possible and pass the responsibility of cloning, copying or converting to the consumer.